pkg/resources/grant_helpers.go

```diff
@@ -64,6 +64,95 @@ type grantID struct {
 	Privilege    string
 }
 
+type stringSet map[string]struct{}
+
+func (ss stringSet) add(s string) {
+	ss[s] = struct{}{}
+}
+
+func (ss stringSet) setEquals(validPrivs []string) bool {
```

Comment:
since setEquals has specific logic for privileges, I wonder if it is worth renaming `stringSet` to `privilegeSet`

Reply:
setEquals might also need a better name, such as `AllPrivsPresent` perhaps